src/views/Timesheets/index.js

```diff
@@ -0,0 +1,539 @@
+import React, { useState } from 'react';
+import 'date-fns';
+import {
+  Grid,
+  Card,
+  CardContent,
+  CardHeader,
+  FormControl,
+  InputLabel,
+  Select,
+  Input,
+  MenuItem,
+  Button,
+  Typography,
+  TextField,
+  Snackbar,
+} from '@material-ui/core/';
+import MuiAlert from '@material-ui/lab/Alert';
+import DateFnsUtils from '@date-io/date-fns';
+import jobs from '../../services/jobs';
+import { MuiPickersUtilsProvider, TimePicker, DatePicker } from '@material-ui/pickers';
+import ShiftDisplay from './components/ShiftDisplay';
+import './timesheets.css';
+
+const Timesheets = (props) => {
+  const [userJobs, setUserJobs] = useState([]);
+  const [selectedDateIn, setSelectedDateIn] = useState(null);
+  const [selectedDateOut, setSelectedDateOut] = useState(null);
+  const [selectedJob, setSelectedJob] = useState(null);
+  const [shiftTooLong, setShiftTooLong] = useState(false);
+  const [shiftTooShort, setShiftTooShort] = useState(false);
+  const [timeOutIsBeforeTimeIn, setTimeOutIsBeforeTimeIn] = useState(false);
+  const [enteredFutureTime, setEnteredFutureTime] = useState(false);
+  const [hoursWorkedInDecimal, setHoursWorkedInDecimal] = useState(0.0);
+  const [userShiftNotes, setUserShiftNotes] = useState('');
+  const [isOverlappingShift, setIsOverlappingShift] = useState(false);
+  const [shiftDisplayComponent, setShiftDisplayComponent] = useState(null);
+  const [snackbarOpen, setSnackbarOpen] = useState(false);
+
+  const handleTimeOutIsBeforeTimeIn = (timeIn, timeOut) => {
+    if (timeIn !== null && timeOut !== null) {
+      let timeDiff = timeOut.getTime() - timeIn.getTime();
+      let calculatedTimeDiff = timeDiff / 1000 / 60 / 60;
+      let roundedHourDifference = (Math.round(calculatedTimeDiff * 4) / 4).toFixed(2);
+      setHoursWorkedInDecimal(roundedHourDifference);
+      let hoursWorked = Math.floor(calculatedTimeDiff);
+      let minutesWorked = Math.round((calculatedTimeDiff - hoursWorked) * 60);
+
+      if (minutesWorked >= 60) {
+        hoursWorked++;
+        minutesWorked = 0;
+      }
+
+      if (timeDiff < 0) {
```

Comment:
Can 54-70 be rewritten to a total of 3 lines using the format `setFunction(lessOrGreaterStatement)`?

Reply:
You're absolutely right, I can't believe we didn't do that in the first place 😆 
Yes, we will make these lines look like `setTimeOutIsBeforeTimeIn(timeDiff < 0)`

Good catch!

Reply:
Thanks, glad to help!